nixos/modules/services/x11/xserver.nix

```diff
@@ -769,9 +767,22 @@ in
             || config.services.greetd.enable
             || config.services.displayManager.ly.enable
             || config.services.displayManager.lemurs.enable
+            || config.services.displayManager.cosmic-greeter.enable
           );
+        hasDesktopEnvironment = (
+          config.services.desktopManager.gnome.enable
+          || config.services.desktopManager.plasma6.enable
+          || config.services.xserver.desktopManager.plasma5.enable
+          || config.services.desktopManager.cosmic.enable
+          || config.services.xserver.desktopManager.xfce.enable
+        );
       in
-      mkIf (default) (mkDefault true);
+      lib.mkIf (noDisplayManager && hasDesktopEnvironment) (
+        lib.warn ''
+          Xserver and a desktop environment is configured but no diplay manager (aka login manager) is enabled.
```

Comment:
This should use module system level `warnings` probably. Also, there should be a way to opt out of the warning, as that's a setup some users actually want.

Also, wording nits: 
- "X server", but generally I'd just say "graphical session", as Wayland sessions are also covered
- "diSplay" manager
- "this system", otherwise the reference is unclear
- "all desktop sessions" (all lower case), though I'd also say "graphical session" here